### PR TITLE
Simplify `cocotb.generators` to use builtins where possible

### PR DESCRIPTION
**Pull request by eric-wieser**
~Also fix `cocotb.generators.byte` to produce bytes and not unicode in Python 3.
Without this change, the generators in `cocotb.generators.packet` produces UTF-8 encoded latin1 (so never actually emits 0xFF, for instance!).~~

---

~~Alternatively...~~ In a future PR should we just remove this submodule?
* It's not documented
* It's somewhat duplicated by the standard library
* ~~`generator.byte` doesn't produce bytes at all on python 3~~
* `generators.packet` has dependencies on `scapy`.

### CODE REVIEW

**Comment by ktbarrett** [comment]
What was the intended use case?

**Comment by eric-wieser** [comment]
Moved the interesting part of this to #1514, as these two pieces need to happen at the same time.

File: cocotb/generators/__init__.py [hunk 1]
```diff
@@ -43,12 +44,7 @@ def repeat(obj, nrepeat=None):
     Kwargs:
         nrepeat (int): The number of times to repeatedly yield obj
     """
-    if nrepeat is None:
-        while True:
-            yield obj
-    else:
-        for i in range(nrepeat):
-            yield obj
+    return itertools.repeat(obj, times=nrepeat)
```
Comment (eric-wieser) [inline cocotb/generators/__init__.py:47]:
```suggestion
    if times is None:
        return itertools.repeat(obj)
    else:
        return itertools.repeat(obj, times=nrepeat)
```

**Comment by ktbarrett** [comment]
> Alternatively... should we just remove this submodule?

This change is trivial enough now I say we bring it in, we can save that for 2.0.

**Comment by eric-wieser** [comment]
I'll probably follow up with a PR to deprecate these two functions.

**Pull request merged**